### ISSUE TITLE
[API] promote org.eclipse.jdt.internal.debug.core.breakpoints.ValidBreakpointLocationLocator into an API

### ISSUE THREAD

**Issue by gayanper** [eclipse-jdt/eclipse.jdt.debug#83]
This classes has usages in vscode java debug plugin to locate elements using the AST tree with given line numbers. providing a better API for this from jdt debug will enable this to be used in such plugins without depending on internals. Also this might be use full for other types of plugins. May be this could be moved to jdt.core since this doesn't have any debug related business logic.

@SarikaSinha WDYT ?

**Comment by SarikaSinha** [comment]
Initially the class existed in debug.ui
org.eclipse.jdt.internal.debug.ui.actions.ValidBreakpointLocationLocator
Which got later moved to debug.core  but was not removed.

I suspect that this might be used from JSDT also even if this is defined in internal package.
I am fine with making it as an API, but moving might break many plug-ins I feel.

**Comment by gayanper** [comment]
@SarikaSinha yes moving might cause issues unless we mirror the class by extending the API. But I’m also fine moving this to debug.core API to start with and add some more APIs such as to get the AST node or binding for matching location. 

I will try to add a API and send a PR
